### ISSUE TITLE
Indicate Search matches entire row

### ISSUE THREAD

**Issue by QueroBartK** [deitch/jstree-grid#179]
currently when searching the matches will be shown in bold red in the jstree but not in the other columns.
I'm not sure if this behaviour is intended or not.

In case it is not you could add the 'jstree-search' class to the columns on the same row as the search result.

This can be done by adding the code : 
```
data.nodes.filter('.jstree-node').each(function(i,node) {
	var nodeId = node.id;
	if(nodeId){
		findDataCell(grid, nodeId).addClass("jstree-search");
	}
});
```
before https://github.com/deitch/jstree-grid/blob/master/jstreegrid.js#L409 (so before getting in the if statement)

(PS: the above code is basically the same as what you do on https://github.com/deitch/jstree-grid/blob/master/jstreegrid.js#L414 but without adding the parent nodes of the matched nodes)

Since the data.nodes to start with only has the matched nodes it will only select the correct rows

In order to unmark the matches we'll need to change the following line https://github.com/deitch/jstree-grid/blob/master/jstreegrid.js#L427 to
`this.gridWrapper.find('div.jstree-grid-cell').removeClass("jstree-search").show();`
so that not only it shows all cells again it also removes all nodes with the jstree-search class

**Comment by deitch** [comment]
Thanks @QueroBartK  for the suggestion.

I am a bit torn on this. It uses that class (and hence the bold red) to indicate a matched cell. I wonder if we should highlight the entire matched row, or only the actual cell that matched (if it matched a cell).

**Comment by QueroBartK** [comment]
@deitch ok, I can understand only highlighting the cell of the matched value. But I wonder how we can know which column originally matched the search query (as we kind of trick jstree in accepting a node when the value has been found in an other column). And if we can't figure out which column was originally matched I feel like it is better to highlight the whole row instead of just the tree node (which might not be the value that was actually matched). Maybe you have some ideas of how we can know the matched column. One idea is including this somehow in the search_callback (but then I guess we'll have to go though all values in a row instead of stopping when a match in that row was found).

**Comment by deitch** [comment]
Well, as you know only too well, the way we do it is use the `search_callback` to indicate if a given row should match or not. Unfortunately, jstree doesn't give us an option to say, "yes, this row matches, but do not highlight it." We could play around with the CSS to get there, but it is asking for trouble.

Given that, yeah, it probably is better to highlight the whole row. Let's do that.

**Comment by deitch** [comment]
Closed in favour of #182
